### PR TITLE
Add developer docs

### PR DESCRIPTION
**Pull request by bhazelton**
Added a bunch of documentation for developers, including instructions for updating the database schema and slightly reorganized the existing computer setup and usage documentation.

To view the new docs, see the RTD build here: https://self-sufficiency-standard--51.org.readthedocs.build/en/51/

Also updated a few docstrings.

### CODE REVIEW

**Comment by bhazelton** [comment]
@s2stewar @cwwannie Bumping this, are these docs clear to you?

File: docs/developer_documentation.md
Comment (s2stewar) [inline docs/developer_documentation.md:1]: A little less clear, but I think it will click more once I attempt this.

**Review by s2stewar — APPROVED** [review]

**Comment by s2stewar** [comment]
Which option is appropriate? Squash and merge? Or rebase and merge? I am leaning towards the former, but am unsure if it matters. @bhazelton

File: docs/developer_documentation.md
Comment (bhazelton) [reply docs/developer_documentation.md:1]: Ok. If there are parts you'd like me to try to clarify let me know, I'm happy to try.

File: docs/computer_setup.md
[the remark this answers is not in the document]
Comment (bhazelton) [reply docs/computer_setup.md:1]: good!

**Pull request merged**